addons/event_sms/models/event_mail.py

```diff
@@ -7,29 +7,35 @@
 class EventTypeMail(models.Model):
     _inherit = 'event.type.mail'
 
-    notification_type = fields.Selection(selection_add=[
-        ('sms', 'SMS')
-    ], ondelete={'sms': 'set default'})
-    sms_template_id = fields.Many2one(
-        'sms.template', string='SMS Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the SMS that will be automatically sent')
-
     @api.model
-    def _get_event_mail_fields_whitelist(self):
-        return super(EventTypeMail, self)._get_event_mail_fields_whitelist() + ['sms_template_id']
+    def _selection_template_model(self):
+        return super(EventTypeMail, self)._selection_template_model() + [('sms.template', 'SMS')]
+
+    notification_type = fields.Selection(selection_add=[('sms', 'SMS')], ondelete={'sms': 'set default'})
+
+    @api.depends('notification_type')
+    def _compute_template_model_id(self):
+        sms_model = self.env['ir.model']._get('sms.template')
+        sms_mails = self.filtered(lambda mail: mail.notification_type == 'sms')
+        sms_mails.template_model_id = sms_model
+        super(EventTypeMail, self - sms_mails)._compute_template_model_id()
 
 
 class EventMailScheduler(models.Model):
     _inherit = 'event.mail'
 
-    notification_type = fields.Selection(selection_add=[
-        ('sms', 'SMS')
-    ], ondelete={'sms': 'set default'})
-    sms_template_id = fields.Many2one(
-        'sms.template', string='SMS Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the SMS that will be automatically sent')
+    @api.model
+    def _selection_template_model(self):
+        return super(EventMailScheduler, self)._selection_template_model() + [('sms.template', 'SMS')]
+
+    notification_type = fields.Selection(selection_add=[('sms', 'SMS')], ondelete={'sms': 'set default'})
+
+    @api.depends('notification_type')
+    def _compute_template_model_id(self):
+        sms_model = self.env['ir.model']._get('sms.template')
+        sms_mails = self.filtered(lambda mail: mail.notification_type == 'sms')
```

Comment:
Could be one-lined also

Reply:
(see comment above, I need the variable `sms_mails`)